pkg/services/cloudmigration/gmsclient/gms_client.go

```diff
@@ -148,8 +158,54 @@ func (c *gmsClientImpl) StartSnapshot(ctx context.Context, session cloudmigratio
 	return &result, nil
 }
 
-func (c *gmsClientImpl) GetSnapshotStatus(context.Context, cloudmigration.CloudMigrationSession, cloudmigration.CloudMigrationSnapshot) (*cloudmigration.CloudMigrationSnapshot, error) {
-	panic("not implemented")
+func (c *gmsClientImpl) GetSnapshotStatus(ctx context.Context, session cloudmigration.CloudMigrationSession, snapshot cloudmigration.CloudMigrationSnapshot) (*cloudmigration.GetSnapshotStatusResponse, error) {
+	c.getStatusMux.Lock()
+	defer c.getStatusMux.Unlock()
+	logger := c.log.FromContext(ctx)
+
+	// Ensure we can't send requests more frequently than supported
+	if !c.getStatusLastQueried.IsZero() && time.Since(c.getStatusLastQueried) < c.minGetStatusPollingPeriod {
```

Comment:
Is this really needed? The end user can just bypass this check by sending the request directly using curl. 

Rate limiting should be done at the receiving end to actually enforce limits

Reply:
Nah, thought I'd be fancy but you're right that it doesn't do much, will remove it to keep things simpler.